### DISCUSSION TITLE
Fx_trading missing full example file

### DISCUSSION THREAD

**Discussion by holguinmora123** [AI4Finance-Foundation/FinRL-Meta#307] [Q&A]
Good morning to all and thank you for sharing your valuable work publicly!:+1:

I've been exploring the files in the meta/env_fx_trading folder on GitHub but have hit a roadblock. Despite my efforts, I haven't been able to get a first example running, and I'm struggling to identify the issue.

Is it possible for you to share the example mentioned at the bottom of the folder description? I'm eager to understand how to apply the model to a test set, but each attempt results in 0 rewards.

The directory mentions a full example at ./neo_finrl/env_fx_trading/env.sh ppo_test.ipynb, but I can't find it. Your guidance would be greatly appreciated.
Thank you so much for your assistance! :pray::pray::pray:

**Comment by julianaddison** [comment]
Hi @holguinmora123, 

If it helps, I got this running after making a number of changes. Managed to run it for one forex pair but am stuck when running it for >1 forex pair. 

I'm running this on Google collab. I had to make some changes to get it running. Listing what helped below,

1. I started with the https://github.com/AI4Finance-Foundation/FinRL-Tutorials/blob/master/1-Introduction/Stock_NeurIPS2018_ElegantRL.ipynb notebook and modified things from there. Ran it till Section 2.1

2. Before Section 2.3, added meta/env_fx_trading into the finRL library. So I copied https://github.com/AI4Finance-Foundation/FinRL-Meta/tree/master/meta/env_fx_trading module into /usr/local/lib/python3.10/dist-packages/finrl/meta/

3. Update the import function in meta/env_fx_trading/enx_fx.py
```
  from finrl.meta.env_fx_trading.util.log_render import render_to_file
  from finrl.meta.env_fx_trading.util.plot_chart import TradingChart
  from finrl.meta.env_fx_trading.util.read_config import EnvConfig
```
4. Ran Section 2.3

5. Download GBPUSD data from https://github.com/cove9988/TradingGym/tree/master

6.  Used the code in https://github.com/AI4Finance-Foundation/FinRL-Tutorials/blob/master/1-Introduction/ForexTrading_Demo.ipynb for the remaining 
'''

For Step 1 and 4. not all import are required but left them as they are to see if I could get things running before cleaning things up later

**Reply by holguinmora123** [reply]
Such a great answer, thank you very much!!!!
